_posts/2024-10-16-18.0.0-release.md

```diff
@@ -0,0 +1,195 @@
+- Improved ordered aggregation performance by reducing complexity from `O(n*m)` to `O(n)`, where `n` is the number of rows and `m` the number of segments in the batch (GH-44052).
+
+### Compute
+
+Casting between string-like and string-view-like types has been implemented (GH-42247).
+
+### Dataset
+
+
+### Filesystems
+
+Writing small files to S3 now uses a single S3 API call instead of three
+(GH-40557). Files larger than 5 MB still go through the regular multipart
+upload mechanism.
+
+Background writes are now implemented and enabled by default for the Azure
+filesystem, dramatically improving the performance of writing to remote files
+(GH-40036).
+
+Finalization of the S3 filesystem layer should hopefully be more robust (GH-44071).
+
+### Gandiva
+
+LLVM 19.1 is now supported (GH-44222).
+
+### GPU
+
+
+### IPC
+
+The seed corpus used for fuzzing the IPC reader has been improved, hopefully
+helping make the IPC reader even more robust against corrupt or malicious
+IPC streams (GH-38041).
+
+### Parquet
+
+A new command line utility `parquet-dump-footer` allows dumping the Thrift-encoded
+footer metadata of a Parquet file, optionally scrubbing confidential data
+(GH-42102). This is part of the effort to collect real-world Parquet metadata
+so as to evaluate the efficiency of future improvements to the Parquet format.
+Please see https://github.com/apache/parquet-benchmark for instructions to submit
+footers representative of your own workloads.
+
+### Substrait
+
+
+## C# notes
+
+- Partial support has been added for LargeBinary, LargeString and LargeList. The column sizes cannot exceed 2 GB in length. (GH-43266).
+- Changes to Flight support were made for better control and compatibility, and to allow Flight Server to be hosted in pre-Kestrel versions of .NET (GH-43907, GH-43672, GH-41347).
+- Support has been added for newly-defined types decimal32 and decimal64 (GH-44271).
+- The import of sliced arrays through the C Data interface now works correctly. (GH-43267)
+## Java notes
+
+**Java 8 is no longer supported.** (GH-38051)
+
+**Gandiva may not work in this release.** For details, please see [GH-43576](https://github.com/apache/arrow/issues/43576).
+
+Basic support for RunEndEncoded was added (GH-39982). The ListView/StringView vector implementations are now more complete, including C Data support (multiple issues).
+
+Several APIs have been updated to accept `long` for addresses in preparation for FFM/large buffer support (GH-43902). We no longer expose `sun.misc.Unsafe` (GH-43479). We no longer ship the `shaded` flight-core JARs (GH-43217).
+
+More options were added to the Dataset ScanOptions API (GH-28866).
+
+## JavaScript notes
+
+
+## Python notes
+
+
+## R notes
+
+For more on what’s in the 18.0.0 R package, see the [R changelog][4].
```

Comment:
```suggestion
## R notes

* R functions that users write that use functions that Arrow supports in dataset
  queries now can be used in queries too. Previously, only functions that used
  arithmetic operators worked.
  For example, `time_hours <- function(mins) mins / 60` worked,
  but `time_hours_rounded <- function(mins) round(mins / 60)` did not;
  now both work. These are automatic translations rather than true user-defined
  functions (UDFs); for UDFs, see `register_scalar_function()`. [GH-41223](https://github.com/apache/arrow/issues/41223)
* `mutate()` expressions can now include aggregations, such as `x - mean(x)`. [GH-41350](https://github.com/apache/arrow/issues/41350)
* `summarize()` supports more complex expressions, and correctly handles cases
  where column names are reused in expressions. [GH-41223](https://github.com/apache/arrow/issues/41223)
* The `na_matches` argument to the `dplyr::*_join()` functions is now supported.
  This argument controls whether `NA` values are considered equal when joining. [GH-41358](https://github.com/apache/arrow/issues/41358)
* R metadata, stored in the Arrow schema to support round-tripping data between
  R and Arrow/Parquet, is now serialized and deserialized more strictly.
  This makes it safer to load data from files from unknown sources into R data.frames. [GH-41969](https://github.com/apache/arrow/issues/41969)
* Turn on the S3 and ZSTD features by default for macOS. [GH-42210](https://github.com/apache/arrow/issues/42210)
* Fix a bug in our implementation of `pull` on grouped datasets, it now
  returns the expected column. [GH-43172](https://github.com/apache/arrow/issues/43172)
  
For full details of what’s in the 18.0.0 R package, see the [R changelog][4].
```

Reply:
@jonkeane these are the 17.0.0 changes though right?

Nic and I are working on a NEWS.md patch in https://github.com/apache/arrow/issues/44496 and we can copy that in here in a bit.